### ISSUE TITLE
Poor documentation

### ISSUE THREAD

**Issue by staboness** [stoplightio/spectral#1799]
**Chore summary**
Improve documentation for JS and add more examples of usage

**Tasks**
After updating to v6, I can't figure out how to register already existing formats, used to be:
```

import { Spectral, Document, Parsers, isOpenApiv2, isOpenApiv3, IRuleResult } from "@stoplight/spectral";

const spectral = new Spectral();  
spectral.registerFormat("oas3", isOpenApiv3);  
spectral.registerFormat("oas2", isOpenApiv2);

```

now: 
```

import { Spectral } from "@stoplight/spectral-core"

const spectral = new Spectral();

spectral.registerFormat <- not existing anymore

```
Unable to use formats in rules, example: 
```

export const oasRule = {
   rules: {
      "exampleRule": {
      "severity": 0,
      "formats": [
        "oas2",
        "oas3"
      ],
      "given": "$.info",
      "message": "Info should contain non empty value",
      "then": {
        "field": "description",
        "function": "truthy"
      }
    }
}
}

```
Receiving error: "exampleRules/formats/0 must be a valid format" and I was unable to find "valid" formats in docs (oas2 and etc. not working)

Unable to extend OAS2 and OAS3 in my rules object:
```
export const oasRule = {
   extends: ["spectral:oas"], // Or OAS3? OAS2? Can't find
   functions: [
    "truthy",
    "xor",
    "pattern",
    "length",
    "casing",
    "schema"
  ],
  // Previous version required to use functions field and exceptions, now its showing me an error below
  rules: {...}
}
```
Gives me an error: no additional properties should be in my rules object.

After removing all "formats" fields out of my rules and "additional properties", received an error: 
`then.function is not a function at Object.lintNode (lintNode.js:21)`
I don't know is it a bug or just I'm doing something wrong.

Example what I did:

```
import { Spectral } from "@stoplight/spectral-core"
import { Parsers } from "@stoplight/spectral-parsers"
import { oasRule } from "../../spectral-rules/oas-rule";

const spectral = new Spectral();

const doc = new Document(code, Parsers.Json); <- `code` here is json formatted openapi 3 (works fine with prev. version)

spectral.setRuleset(oasRule); <- this rules is from .js file
spectral.run(doc).then(response => console.log(response)); <- getting error "then.function is not a function"
```
How to extend common OAS rules to check with my custom rules? How to specify a format for a specific rule?

**Comment by P0lip** [comment]
Hey!
You no longer need to register formats.
They come with rulesets now, so whichever format ruleset specifies is going to be respected by Spectral.
This means, you can get rid of `registerFormat` call entirely.
This also lets you override these formats if you'd like to do so.

We have an updated JS guide section over [here](https://github.com/stoplightio/spectral/blob/develop/docs/guides/3-javascript.md#linting-a-yaml-string).

You can still continue to use JSON/YAML rulesets in your JS code, but you'll only need to use the [migrator](https://github.com/stoplightio/spectral/tree/develop/packages/ruleset-migrator#programmatically) prior to the import.
Then, once you import that JS file, you can pass it over to `setRuleset` method.

If you do intend to use the JS format, here's an example ruleset:

```js
import { oas as spectralOas } from '@stoplight/spectral-rulesets';
import { truthy, xor, pattern, length, casing, schema } from '@stoplight/spectral-functions';
import { oas2, oas3 } from '@stoplight/spectral-formats';

export const oasRule = {
   extends: [spectralOas],
   formats: [oas2, oas3],
   // functions is no longer needed, you load them explicitly now as any other member
   rules: {
    'my-rule': {
      given: '$',
      then: {
         function: truthy,
       }
    },
    'my-rule2': {
      given: '$',
      then: {
         function: pattern,
         functionOptions: {
           match: '^[a-z]+$',
         }
       }
    }
  }
}
```

**Comment by staboness** [comment]
> Hey!
> 
> You no longer need to register formats.
> 
> They come with rulesets now, so whichever format ruleset specifies is going to be respected by Spectral.
> 
> This means, you can get rid of `registerFormat` call entirely.
> 
> This also lets you override these formats if you'd like to do so.
> 
> 
> 
> We have an updated JS guide section over [here](https://github.com/stoplightio/spectral/blob/develop/docs/guides/3-javascript.md#linting-a-yaml-string).
> 
> 
> 
> You can still continue to use JSON/YAML rulesets in your JS code, but you'll only need to use the [migrator](https://github.com/stoplightio/spectral/tree/develop/packages/ruleset-migrator#programmatically) prior to the import.
> 
> Then, once you import that JS file, you can pass it over to `setRuleset` method.
> 
> 
> 
> If you do intend to use the JS format, here's an example ruleset:
> 
> 
> 
> ```js
> 
> import { oas as spectralOas } from '@stoplight/spectral-rulesets';
> 
> import { truthy, xor, pattern, length, casing, schema } from '@stoplight/spectral-functions';
> 
> import { oas2, oas3 } from '@stoplight/spectral-formats';
> 
> 
> 
> export const oasRule = {
> 
>    extends: [spectralOas],
> 
>    formats: [oas2, oas3],
> 
>    // functions is no longer needed, you load them explicitly now as any other member
> 
>    rules: {
> 
>     'my-rule': {
> 
>       given: '$',
> 
>       then: {
> 
>          function: truthy,
> 
>        }
> 
>     },
> 
>     'my-rule2': {
> 
>       given: '$',
> 
>       then: {
> 
>          function: pattern,
> 
>          functionOptions: {
> 
>            match: '^[a-z]+$',
> 
>          }
> 
>        }
> 
>     }
> 
>   }
> 
> }
> 
> ```
> 
> 
> 
> 

Great! Thank you for fast response and providing additional example!
